bundles/org.openhab.voice.rustpotterks/src/main/java/org/openhab/voice/rustpotterks/internal/RustpotterKSService.java

```diff
@@ -62,27 +66,41 @@
         + " Keyword Spotter", description_uri = SERVICE_CATEGORY + ":" + SERVICE_ID)
 public class RustpotterKSService implements KSService {
     private static final String RUSTPOTTER_FOLDER = Path.of(OpenHAB.getUserDataFolder(), "rustpotter").toString();
+    private static final String RUSTPOTTER_RECORDS_FOLDER = Path.of(RUSTPOTTER_FOLDER, "records").toString();
     private final Logger logger = LoggerFactory.getLogger(RustpotterKSService.class);
-    private final ScheduledExecutorService executor = ThreadPoolManager.getScheduledPool("OH-voice-rustpotterks");
+    private final ExecutorService executor = ThreadPoolManager.getPool("voice-rustpotterks");
     private RustpotterKSConfiguration config = new RustpotterKSConfiguration();
+    private final List<RustpotterMutex> runningInstances = new ArrayList<>();
     static {
         Logger logger = LoggerFactory.getLogger(RustpotterKSService.class);
-        File directory = new File(RUSTPOTTER_FOLDER);
-        if (!directory.exists()) {
-            if (directory.mkdir()) {
-                logger.info("rustpotter dir created {}", RUSTPOTTER_FOLDER);
+        tryCreateDir(RUSTPOTTER_FOLDER, logger, "rustpotter dir created {}");
+        tryCreateDir(RUSTPOTTER_RECORDS_FOLDER, logger, "rustpotter record dir created {}");
+    }
+
+    private static void tryCreateDir(String rustpotterFolder, Logger logger, String msg) {
+        File addonDir = new File(rustpotterFolder);
+        if (!addonDir.exists()) {
+            if (addonDir.mkdir()) {
+                logger.info(msg, rustpotterFolder);
             }
         }
```

Comment:
As you already use `java.nio.file.Path` you might as well stop using `java.io.File` here as well and replace it with `Files.createDirectories`. In general using `java.nio` is preferred over `java.io.File` because it has nicer APIs and better performance.

Reply:
Thank you for the suggestion, I wasn't aware of that, I have moved the code to use Path instances and the helpers of the Files class, and as you said it looks much better.

One side question about this, more related to the service design. Can I move the folder creation from the static context to the activate method and throw the exceptions to prevent the add-on from been loaded if the folder creation doesn't succeed? I don't know if is correct to throw exceptions from the activate method or if those should be catch inside it.

Reply:
Creating the folders in the activate method is probably better so they will be recreated (when missing) every time the bundle is restarted. If thrown exceptions result in big stacktraces it might be better to catch exceptions and log a helpful message instead.